File: trash/flores_gomez_javier_alberto/libro.py
import pandas as pd


class Libro:
    def __init__(self, titulo: str, autor: str, genero: str, cantidad_copias, isbn):
        self._titulo = titulo
        self._autor = autor
        self._genero = genero
        self._cantidad_copias = cantidad_copias
        self._isbn = isbn
        if cantidad_copias != 0:
            self._estado = "Disponible"
        else:
            self._estado = "Ocupado"

    @property
    def titulo(self):
        return self._titulo

    @property
    def autor(self):
        return self._autor

    @property
    def genero(self):
        return self._genero

    @property
    def cantidad_copias(self):
        return self._cantidad_copias

    @property
    def isbn(self):
        return self._isbn

    @property
    def estado(self):
        return self._estado

    def _cambiar_estado(self, dataframe: pd.DataFrame, index):
        if self._estado == "Disponible":
            self._cantidad_copias -= 1
            dataframe.iloc[index, 3] = self._cantidad_copias
            dataframe.to_csv("list_of_books.csv", index=False)
            if self._cantidad_copias == 0:
                self._estado = "Ocupado"

    def _devolver_libro(self, dataframe: pd.DataFrame, index):
        self._cantidad_copias += 1
        dataframe.iloc[index, 3] = self._cantidad_copias
        dataframe.to_csv("list_of_books.csv", index=False)
        if self._estado == "Ocupado":
            self._estado = "Disponible"
            
    def mostrar_libro(self):
        print(f"Titulo: {self._titulo}")
        print(f"Autor: {self._autor}")
        print(f"Genero: {self._genero}")
        print(f"ISBN: {self._isbn}")

File: development_(codes)/src/edd_proyecto/Libro.java
package edd_proyecto;

import javax.swing.JOptionPane;
//import joinery.DataFrame;
public class Libro {
	private String titulo;
	private String autor;
	private String genero;
	private Integer cantidad_copias;
	private String isbn;
	private String estado;
	
	
	
	public Libro(String titulo, String autor, String genero, Integer cantidad_copias, String isbn) {
		this.titulo = titulo;
		this.autor = autor;
		this.genero = genero;
		this.cantidad_copias = cantidad_copias;
		this.isbn = isbn;
		  if (cantidad_copias != 0) {
	            this.setEstado("Disponible");
		  } else {
	            this.setEstado("Ocupado");
		  }
	}
	public String getTitulo() {
		return titulo;
	}
	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}
	public String getAutor() {
		return autor;
	}
	public void setAutor(String autor) {
		this.autor = autor;
	}
	public String getGenero() {
		return genero;
	}
	public void setGenero(String genero) {
		this.genero = genero;
	}
	public Integer getCantidad_copias() {
		return cantidad_copias;
	}
	public void setCantidad_copias(Integer cantidad_copias) {
		this.cantidad_copias = cantidad_copias;
	}
	public String getIsbn() {
		return isbn;
	}
	public void setIsbn(String isbn) {
		this.isbn = isbn;
	}
	public String getEstado() {
		return estado;
	}
	public void setEstado(String estado) {
		this.estado = estado;
	}
	public void _cambiar_estado(/*dataframe: pd.DataFrame*/int index ) {
		if(estado=="Disponible") {
			cantidad_copias-=1;
			//dataframe.iloc[index, 3] = cantidad_copias;
			//dataframe.to_csv("list_of_books.csv", index=False);
			
			if(cantidad_copias==0) {
				estado = "Ocupado";
				
			}
			
		}
	}
	
	public void _devolver_libro(/*self, dataframe: pd.DataFrame*/ int index) {
		cantidad_copias += 1;
		 //dataframe.iloc[index, 3] = self._cantidad_copias
		 //dataframe.to_csv("list_of_books.csv", index=False)
		
		if(estado=="Ocupado") {
			estado = "Disponible";
		}
		
	}
	
	public void mostrar_libro() {
		JOptionPane.showMessageDialog(null, "\n Titulo: "+titulo);
		JOptionPane.showMessageDialog(null, "\n Autor: "+autor);
		JOptionPane.showMessageDialog(null, "\n Género: "+genero);
		JOptionPane.showMessageDialog(null, "\n ISBN: "+isbn);

	}
	
}
